Sources/TuistKit/Services/RunService.swift

```diff
@@ -0,0 +1,142 @@
+
+    var description: String {
+        switch self {
+        case let .appNotFound(targetName):
+            return "Couldn't find any apps built for \(targetName)"
+        case .simulatorNotFound:
+            return "Could't find simulator"
+        case let .targetNotFound(forSchemeName):
+            return "Couldn't find \(forSchemeName)"
+        }
+    }
+
+    var type: ErrorType {
+        switch self {
+        case .appNotFound, .simulatorNotFound, .targetNotFound:
+            return .abort
+        }
+    }
+}
+
+final class RunService {
+    let simulatorController: SimulatorControlling
+
+    let buildGraphInspector: BuildGraphInspecting
+
+    let projectGenerator: ProjectGenerating
+
+    let xcodebuildController: XcodeBuildControlling
+
+    var appPath: AbsolutePath?
+
+    var buildableTarget: Target?
+
+    init(simulatorController: SimulatorControlling = SimulatorController(),
+         buildGraphInspector: BuildGraphInspecting = BuildGraphInspector(),
+         projectGenerator: ProjectGenerating = ProjectGenerator(),
+         xcodebuildController: XcodeBuildControlling = XcodeBuildController())
+    {
+        self.simulatorController = simulatorController
+        self.buildGraphInspector = buildGraphInspector
+        self.projectGenerator = projectGenerator
+        self.xcodebuildController = xcodebuildController
+    }
+
+    func run(schemeName: String) throws {
+        try buildScheme(schemeName)
+
+        guard let buildableTarget = buildableTarget else {
+            throw RunServiceError.targetNotFound(forSchemeName: schemeName)
+        }
+
+        appPath = try findApp(for: buildableTarget)
+        try launchApp(for: buildableTarget)
+    }
+
+    private func buildScheme(_ schemeName: String) throws {
+        let graph = try projectGenerator.load(path: FileHandler.shared.currentPath)
+        guard let scheme = buildGraphInspector.buildableSchemes(graph: graph).first(where: { $0.name == schemeName }) else {
+            throw RunServiceError.appNotFound(targetName: "")
+        }
+
+        buildableTarget = buildGraphInspector.buildableTarget(scheme: scheme, graph: graph)
+
+        let workspacePath = try buildGraphInspector.workspacePath(directory: FileHandler.shared.currentPath)!
+
+        _ = try xcodebuildController.build(.workspace(workspacePath),
+                                           scheme: schemeName,
+                                           clean: false,
+                                           arguments: buildGraphInspector.buildArguments(target: buildableTarget!,
+                                                                                         configuration: nil))
+            .printFormattedOutput()
+            .toBlocking()
+            .last()
+    }
+
+    private func findApp(for target: Target) throws -> AbsolutePath {
+        let appsLocation = FileHandler.shared.locateDirectory("../Library/Developer/Xcode/DerivedData/",
```

Comment:
I'd extend [XcodeController](https://github.com/tuist/tuist/blob/master/Sources/TuistSupport/Xcode/XcodeController.swift) with a method to get the derive data directory:

```
func derivedDataPath() throws -> AbsolutePath
```

Then, we could follow the logic described in [this blog post](https://pewpewthespells.com/blog/xcode_build_locations.html):
1. If custom location is defined, take the custom location.
2. Otherwise, read the default value `~/Library/Developer/Xcode/DerivedData`.